pkg/controller/servicebindingrequest/retriever.go

```diff
@@ -134,9 +155,29 @@ func (r *Retriever) extractConfigMapItemName(xDescriptor string) string {
 	return strings.ReplaceAll(xDescriptor, fmt.Sprintf("%s:", configMapPrefix), "")
 }
 
+// markVisitedPaths updates all visited paths in Cache, This initializes the Cache map
+func (r *Retriever) markVisitedPaths(name, keyPath, fromPath string) {
+	if _, ok := r.Cache[fromPath]; !ok {
+		r.Cache[fromPath] = make(map[string]interface{})
+	}
+	if _, ok := r.Cache[fromPath].(map[string]interface{})[name]; !ok {
+		r.Cache[fromPath].(map[string]interface{})[name] = make(map[string]interface{})
+	}
+	if _, ok := r.Cache[fromPath].(map[string]interface{})[name].(map[string]interface{}); !ok {
+		r.Cache[fromPath].(map[string]interface{})[name] = make(map[string]interface{})
+	}
+	if _, ok := r.Cache[fromPath].(map[string]interface{})[name].(map[string]interface{})[keyPath]; !ok {
+		r.Cache[fromPath].(map[string]interface{})[name].(map[string]interface{})[keyPath] = make(map[string]interface{})
+	}
+}
+
 // readSecret based in secret name and list of items, read a secret from the same namespace informed
 // in plan instance.
-func (r *Retriever) readSecret(name string, items []string) error {
+func (r *Retriever) readSecret(
+	name string,
+	items []string,
+	fromPath string,
+	path string) error {
```

Comment:
@akashshinde, @Avni-Sharma, I see in many places formatting errors, like in `error` here. Could you double check if `go format` is working as expected, as in, every time a file is saved?

Reply:
@otaviof I don't see any error with formatting and `make lint` passed too